tests/base_scraper/test_base_scraper.py

```diff
@@ -5,133 +5,120 @@
-                                            destination=destination,
-                                            logger=self.logger)
-        self.assertEqual(scraper.filename, destination)
-
-        # destination only has filename
-        scraper = mozdownload.DirectScraper(url=test_url,
-                                            destination=filename,
-                                            logger=self.logger)
-        self.assertEqual(scraper.filename, os.path.abspath(filename))
-
-        # destination directory does not exist
-        destination = os.path.join(self.temp_dir, 'temp_folder', filename)
-        scraper = mozdownload.DirectScraper(url=test_url,
-                                            destination=destination,
-                                            logger=self.logger)
-        self.assertEqual(scraper.destination, destination)
-
-        # ensure that multiple non existing directories are created
-        destination = os.path.join(self.temp_dir, 'tmp1', 'tmp2', filename)
-        scraper = mozdownload.DirectScraper(url=test_url,
-                                            destination=destination,
-                                            logger=self.logger)
-        self.assertEqual(scraper.destination, destination)
+
+@pytest.mark.skip(reason="Permanent failure due to mozqa.com not available anymore (#492)")
+def test_valid_authentication(tmpdir):
+    """testing with valid authentication"""
+    username = 'mozilla'
+    password = 'mozilla'
+    basic_auth_url = 'http://mozqa.com/data/mozqa.com/http_auth/basic/'
+    scraper = mozdownload.DirectScraper(destination=str(tmpdir),
+                                        url=basic_auth_url,
+                                        username=username,
+                                        password=password)
+    scraper.download()
+    assert os.path.isfile(os.path.join(str(tmpdir), 'mozqa.com'))
+
+def test_destination_isdirectory(httpd, tmpdir):
+    """destination is directory"""
+    filename = 'download_test.txt'
+    test_url = urljoin(httpd.get_url(), filename)
+    scraper = mozdownload.DirectScraper(url=test_url, destination=str(tmpdir))
+    assert scraper.filename == os.path.join(str(tmpdir), filename)
+
+
+def test_destination_hasdirectory(httpd, tmpdir):
+    """destination has directory path with filename"""
+    filename = 'download_test.txt'
+    test_url = urljoin(httpd.get_url(), filename)
+    destination = os.path.join(str(tmpdir), filename)
+    scraper = mozdownload.DirectScraper(url=test_url, destination=destination)
+    assert scraper.filename == destination
+
+def test_destination_hasfile(httpd):
+    """destination only has filename"""
+    filename = 'download_test.txt'
+    test_url = urljoin(httpd.get_url(), filename)
+    scraper = mozdownload.DirectScraper(url=test_url, destination=filename)
+    assert scraper.filename == os.path.abspath(filename)
+
+def test_destination_doesnotexist(httpd, tmpdir):
```

Comment:
Please separate each word with an underscore.

Reply:
I was worried it would make function names too big. But will follow the exact pep standards keeping the function names concise.